exp/api/v1alpha3/awsmanagedcontrolplane_types.go

```diff
@@ -0,0 +1,184 @@
+	// permission to make API calls. If no name is supplied
+	// then a role is created. If the role is pre-existing
+	// we will treat it as unmanaged and not delete it on
+	// deletion
+	// +kubebuilder:validation:MinLength:=2
+	// +optional
+	RoleName *string `json:"roleName"`
+
+	// RoleAdditionalPolicies allows you to attach additional polices to
+	// the control plane role.
+	// +optional
+	RoleAdditionalPolicies *[]string `json:"roleAdditionalPolicies"`
+
+	// Logging specifies which EKS Cluster logs should be enabled. Entries for
+	// each of the enabled logs will be sent to CloudWatch
+	// +optional
+	Logging *ControlPlaneLoggingSpec `json:"logging,omitempty"`
+
+	// EncryptionConfig specifies the encryption configuration for the cluster
+	// +optional
+	EncryptionConfig *EncryptionConfig `json:"encryptionConfig,omitempty"`
+
+	// AdditionalTags is an optional set of tags to add to AWS resources managed by the AWS provider, in addition to the
+	// ones added by default.
+	// +optional
+	AdditionalTags infrav1.Tags `json:"additionalTags,omitempty"`
+
+	// Endpoints specifies access to this cluster's control plane endpoints
+	// +optional
+	EndpointAccess EndpointAccess `json:"endpointAccess,omitempty"`
+
+	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
+	// +optional
+	ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+
+	// Bastion contains options to configure the bastion host.
+	// +optional
+	Bastion infrav1.Bastion `json:"bastion"`
+
+	// TokenMethod is used to specify the method for obtaining a client token for communicating with EKS
+	// iam-authenticator - obtains a client token using iam-authentictor
+	// aws-cli - obtains a client token using the AWS CLI
+	// Defaults to iam-authenticator
+	// +kubebuilder:default=iam-authenticator
+	// +kubebuilder:validation:Enum=iam-authenticator;aws-cli
+	TokenMethod *EKSTokenMethod `json:"tokenMethod,omitempty"`
+}
+
+// EndpointAccess specifies how control plane endpoints are accessible
+type EndpointAccess struct {
+	// Public controls whether control plane endpoints are publicly accessible
+	// +optional
+	Public *bool `json:"public,omitempty"`
+	// PublicCIDRs specifies which blocks can access the public endpoint
+	// +optional
+	PublicCIDRs []*string `json:"publicCIDRs,omitempty"`
+	// Private points VPC-internal control plane access to the private endpoint
+	// +optional
+	Private *bool `json:"private,omitempty"`
+}
+
+// EncryptionConfig specifies the encryption configuration for the EKS clsuter
+type EncryptionConfig struct {
+	// Provider specifies the ARN or alias of the CMK (in AWS KMS)
+	Provider *string `json:"provider,omitempty"`
+	//Resources specifies the resources to be encrypted
+	Resources []*string `json:"resources,omitempty"`
+}
+
+// AWSManagedControlPlaneStatus defines the observed state of AWSManagedControlPlane
+type AWSManagedControlPlaneStatus struct {
```

Comment:
I think we also need an `Initialized` in here:

https://github.com/kubernetes-sigs/cluster-api/blob/e955160f6f30f61eabce65231899a1fe9d513046/controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_types.go#L103-L106

https://github.com/kubernetes-sigs/cluster-api/blob/f245e8ac13677b8ad2e7079924d0552cc944b5a7/controllers/cluster_controller_phases.go#L230-L238

I don't know if this actually gives us any value though. Not sure if the distinction between ready and initialized means anything to us in EKS land.

Reply:
If there's no value, I'll update the EKSConfig PR to not check for it.

Reply:
Thats an interesting one. I did have `Initialized` in the status originally but took it out as there was no distinction from `Ready` for the managed control plane. I can add it it back in.

Having a look it seems i have also lost somewhere along the way setting `Ready`.

@rudoi i'll add Initialized in (and restore the setting of `Ready`

Reply:
Added Initialize